libs/pubsub/query/query.go

```diff
@@ -301,87 +375,94 @@ func matchValue(value string, op Operator, operand reflect.Value) bool {
 			v, err = time.Parse(DateLayout, value)
 		}
 		if err != nil {
-			panic(fmt.Sprintf("failed to convert value %v from tag to time.Time: %v", value, err))
+			return false, fmt.Errorf("failed to convert value %v from event attribute to time.Time: %w", value, err)
 		}
 
 		switch op {
 		case OpLessEqual:
-			return v.Before(operandAsTime) || v.Equal(operandAsTime)
+			return (v.Before(operandAsTime) || v.Equal(operandAsTime)), nil
 		case OpGreaterEqual:
-			return v.Equal(operandAsTime) || v.After(operandAsTime)
+			return (v.Equal(operandAsTime) || v.After(operandAsTime)), nil
 		case OpLess:
-			return v.Before(operandAsTime)
+			return v.Before(operandAsTime), nil
 		case OpGreater:
-			return v.After(operandAsTime)
+			return v.After(operandAsTime), nil
 		case OpEqual:
-			return v.Equal(operandAsTime)
+			return v.Equal(operandAsTime), nil
 		}
 
 	case reflect.Float64:
-		operandFloat64 := operand.Interface().(float64)
 		var v float64
 
+		operandFloat64 := operand.Interface().(float64)
+		filteredValue := numRegex.FindString(value)
+
 		// try our best to convert value from tags to float64
-		v, err := strconv.ParseFloat(value, 64)
+		v, err := strconv.ParseFloat(filteredValue, 64)
 		if err != nil {
-			panic(fmt.Sprintf("failed to convert value %v from tag to float64: %v", value, err))
+			return false, fmt.Errorf("failed to convert value %v from event attribute to float64: %w", filteredValue, err)
 		}
 
 		switch op {
 		case OpLessEqual:
-			return v <= operandFloat64
+			return v <= operandFloat64, nil
 		case OpGreaterEqual:
-			return v >= operandFloat64
+			return v >= operandFloat64, nil
 		case OpLess:
-			return v < operandFloat64
+			return v < operandFloat64, nil
 		case OpGreater:
-			return v > operandFloat64
+			return v > operandFloat64, nil
 		case OpEqual:
-			return v == operandFloat64
+			return v == operandFloat64, nil
 		}
 
 	case reflect.Int64:
-		operandInt := operand.Interface().(int64)
 		var v int64
+
+		operandInt := operand.Interface().(int64)
+		filteredValue := numRegex.FindString(value)
+
 		// if value looks like float, we try to parse it as float
-		if strings.ContainsAny(value, ".") {
-			v1, err := strconv.ParseFloat(value, 64)
+		if strings.ContainsAny(filteredValue, ".") {
+			v1, err := strconv.ParseFloat(filteredValue, 64)
 			if err != nil {
-				panic(fmt.Sprintf("failed to convert value %v from tag to float64: %v", value, err))
+				return false, fmt.Errorf("failed to convert value %v from event attribute to float64: %w", filteredValue, err)
 			}
+
 			v = int64(v1)
 		} else {
 			var err error
 			// try our best to convert value from tags to int64
-			v, err = strconv.ParseInt(value, 10, 64)
+			v, err = strconv.ParseInt(filteredValue, 10, 64)
 			if err != nil {
-				panic(fmt.Sprintf("failed to convert value %v from tag to int64: %v", value, err))
+				return false, fmt.Errorf("failed to convert value %v from event attribute to int64: %w", filteredValue, err)
```

Comment:
```suggestion
				return false, errors.Wrapf(err, "failed to convert value %v from event attribute to int64", filteredValue)
```

Reply:
Why this over what Go 1.13 recommends?

ref: https://blog.golang.org/go1.13-errors